velox/vector/tests/VariantToVectorTest.cpp

```diff
@@ -30,11 +30,21 @@ class VariantToVectorTest : public testing::Test {
 
  public:
   template <TypeKind KIND>
-  void testCreateVector(const std::vector<variant>& inputArray) {
+  void testCreateVector(
+      const std::vector<variant>& inputArray,
+      std::optional<TypePtr> type) {
     using TCpp = typename TypeTraits<KIND>::NativeType;
 
     auto varArray = variant::array(inputArray);
-    auto arrayVector = variantArrayToVector(varArray.array(), pool_.get());
+    constexpr bool isDecimalType =
+        (KIND == facebook::velox::TypeKind::SHORT_DECIMAL ||
+         KIND == facebook::velox::TypeKind::LONG_DECIMAL);
+    ArrayVectorPtr arrayVector;
+    if constexpr (isDecimalType) {
+      arrayVector = variantArrayToVector(varArray.array(), pool_.get(), type);
+    } else {
+      arrayVector = variantArrayToVector(varArray.array(), pool_.get(), type);
+    }
```

Comment:
I don't see why `if block` is needed here. The variantArrayToVector call seems to be same.

Reply:
Reverted, apologies for the oversight.